### PR TITLE
Provides Symfony 4 compatibility

### PR DESCRIPTION
**Pull request by romain-pierre**
Hi, I'm working to get a compatible version of this bundle for the forthcoming Symfony 4. Don't hesitate to suggest fixes, leave comments about this PR.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,14 +14,14 @@
...
+        "symfony/yaml": "^2.7 || ^3.0 || ^4.0"
     },
     "require-dev": {
         "doctrine/cache": "1.*",
         "predis/predis": "^1.0",
-        "symfony/console": "^2.7 || ^3.0",
-        "symfony/phpunit-bridge": "^2.7 || ^3.0",
+        "symfony/console": "^2.7 || ^3.0 || ^4.0",
+        "symfony/phpunit-bridge": "^2.7 || ^3.0 || ^4.0",
         "phpunit/phpunit": "4.8.*"
```
Comment (xabbuh) [inline composer.json]: the version constraint here should now be `^4.8.35 || ^5.4.3 || ^6.0`

Comment (romain-pierre) [reply composer.json]: Yeah, good idea!

**Comment by snc** [comment]
I've seen you removed the WIP status, is this ready for merging?

File: .travis.yml [hunk 2]
```diff
@@ -8,16 +8,19 @@ php:
...
-      env: SYMFONY_VERSION="2.7.*"
+      env: SYMFONY_VERSION="2.7.*" SYMFONY_DEPRECATIONS_HELPER="weak"
     - php: 5.6
-      env: SYMFONY_VERSION="2.8.*" SYMFONY_DEPRECATIONS_HELPER="strict"
+      env: SYMFONY_VERSION="2.8.*"
     - php: 5.6
-      env: SYMFONY_VERSION="3.0.*" SYMFONY_DEPRECATIONS_HELPER="strict"
+      env: SYMFONY_VERSION="3.0.*"
+    - php: 7.1
+      env: SYMFONY_VERSION="4.0.0-BETA3"
```
Comment (SiM07) [inline .travis.yml]: To use latest BETA (today 4.0.0-BETA4) version :
`env: DEPENDENCIES=beta`

and `before_install` :

```
- if [ "$DEPENDENCIES" = "beta" ]; then composer config minimum-stability beta; fi;
```

**Comment by romain-pierre** [comment]
@snc I'm waiting for some comments or approbations because I've not checked all features of this bundle with Symfony 4. It's seem ok to me but maybe not for somebody else. If you think it's ok, you can merge.

@SiM07 Thank you, I will try to update this PR tonight.

**Comment by snc** [comment]
I've no comments to add :-) Would be nice if you apply the proposed changes from @SiM07 then I'll merge this.

**Comment by romain-pierre** [comment]
Done! I've updated.

File: Resources/config/redis.xml [hunk 3]
```diff
@@ -16,6 +16,12 @@
             <tag name="data_collector" template="@SncRedis/Collector/redis.html.twig" id="redis" />
             <argument type="service" id="snc_redis.logger" />
         </service>
+        <service id="snc_redis.command.flush_all" class="Snc\RedisBundle\Command\RedisFlushallCommand" public="false">
+            <tag name="console.command" />
```
Comment (bgaleotti) [inline Resources/config/redis.xml]:
```xml
<tag name="console.command" command="redis:flushall" />
```
in order to be lazy-loaded in Symfony 3.4+

File: Resources/config/redis.xml [hunk 4]
```diff
@@ -16,6 +16,12 @@
...
+        </service>
+        <service id="snc_redis.command.flush_db" class="Snc\RedisBundle\Command\RedisFlushdbCommand" public="false">
+            <tag name="console.command" />
```
Comment (bgaleotti) [inline Resources/config/redis.xml]:
```xml
<tag name="console.command" command="redis:flushdb" />
```

**Comment by cordoval** [comment]
@romain-pierre sf4 is out already, this is all is left i guess and then \LGTM

File: .travis.yml [hunk 5]
```diff
@@ -8,16 +8,19 @@ php:
...
-      env: SYMFONY_VERSION="2.7.*"
+      env: SYMFONY_VERSION="2.7.*" SYMFONY_DEPRECATIONS_HELPER="weak"
     - php: 5.6
-      env: SYMFONY_VERSION="2.8.*" SYMFONY_DEPRECATIONS_HELPER="strict"
+      env: SYMFONY_VERSION="2.8.*"
     - php: 5.6
-      env: SYMFONY_VERSION="3.0.*" SYMFONY_DEPRECATIONS_HELPER="strict"
+      env: SYMFONY_VERSION="3.0.*"
+    - php: 7.1
+      env: SYMFONY_VERSION="4.0.*" MINIMUM_STABILITY="beta"
```
Comment (cordoval) [inline .travis.yml]: you can remove this beta thingy

File: .travis.yml [hunk 6]
```diff
@@ -29,13 +32,14 @@ cache:
...
   global:
-    - SYMFONY_DEPRECATIONS_HELPER="weak"
+    - SYMFONY_DEPRECATIONS_HELPER="strict"
 
 before_install:
   - phpenv config-rm xdebug.ini || true
   - if [[ $TRAVIS_PHP_VERSION != hhvm ]]; then echo "extension = redis.so" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi;
   - if [[ $TRAVIS_PHP_VERSION != hhvm ]]; then phpenv rehash; fi;
   - composer self-update
+  - if [ "$MINIMUM_STABILITY" != "" ]; then composer config minimum-stability ${MINIMUM_STABILITY}; fi;
```
Comment (cordoval) [inline .travis.yml]: this can be removed

**Comment by romain-pierre** [comment]
Updated and ready.

**Pull request merged**